pkg/component/etcd/configmap/configmap.go

```diff
@@ -0,0 +1,208 @@
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package configmap
+
+import (
+	"context"
+	"fmt"
+	"strconv"
+
+	druidv1alpha1 "github.com/gardener/etcd-druid/api/v1alpha1"
+	corev1 "k8s.io/api/core/v1"
+
+	"github.com/gardener/gardener/pkg/controllerutils"
+	gardenercomponent "github.com/gardener/gardener/pkg/operation/botanist/component"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/utils/pointer"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+)
+
+type component struct {
+	client    client.Client
+	namespace string
+
+	values Values
+}
+
+// New creates a new configmap deployer instance.
+func New(c client.Client, namespace string, values Values) gardenercomponent.Deployer {
+	return &component{
+		client:    c,
+		namespace: namespace,
+		values:    values,
+	}
+}
+
+func (c *component) Deploy(ctx context.Context) error {
+	cm := c.emptyConfigmap(c.values.ConfigMapName)
+
+	if err := c.syncConfigmap(ctx, cm); err != nil {
+		return err
+	}
+
+	return nil
+}
+
+func (c *component) syncConfigmap(ctx context.Context, cm *corev1.ConfigMap) error {
+	_, err := controllerutils.GetAndCreateOrMergePatch(ctx, c.client, cm, func() error {
+		etcdConfigYaml := `# Human-readable name for this member.
+    name: ` + fmt.Sprintf("etcd-%s", c.values.EtcdUID[:6]) + `
+    # Path to the data directory.
+    data-dir: /var/etcd/data/new.etcd`
+		metricsLevel := druidv1alpha1.Basic
+		if c.values.Metrics != nil {
+			metricsLevel = *c.values.Metrics
+		}
+
+		etcdConfigYaml = etcdConfigYaml + `
+    # metrics configuration
+    metrics: ` + string(metricsLevel) + `
+    # Number of committed transactions to trigger a snapshot to disk.
+    snapshot-count: 75000
+
+    # Accept etcd V2 client requests
+    enable-v2: false`
+		var quota int64 = 8 * 1024 * 1024 * 1024 // 8Gi
+		if c.values.Quota != nil {
+			quota = c.values.Quota.Value()
+		}
+
+		etcdConfigYaml = etcdConfigYaml + `
+    # Raise alarms when backend size exceeds the given quota. 0 means use the
+    # default quota.
+    quota-backend-bytes: ` + fmt.Sprint(quota)
+
+		enableTLS := (c.values.TLS != nil)
+
+		protocol := "http"
+		if enableTLS {
+			protocol = "https"
+		}
+
+		clientPort := strconv.Itoa(int(pointer.Int32Deref(c.values.ClientPort, defaultClientPort)))
+
+		serverPort := strconv.Itoa(int(pointer.Int32Deref(c.values.ServerPort, defaultServerPort)))
+
+		etcdConfigYaml = etcdConfigYaml + `
+    # List of comma separated URLs to listen on for client traffic.
+    listen-client-urls: ` + protocol + `://0.0.0.0:` + clientPort + `
+
+    # List of this member's client URLs to advertise to the public.
+    # The URLs needed to be a comma-separated list.
+    advertise-client-urls: ` + c.values.PeerServiceName + `@` + c.values.EtcdNameSpace + `@` + clientPort + `
```

Comment:
Values like `c.values.PeerServiceName` are never acutally used in the final (downloaded) ConfigMap because they are [replaced](https://github.com/gardener/etcd-backup-restore/pull/419/files#diff-3a74c5f616b5783bc51c10a48dcde0da7aa4d0e3b814070bf3d71eb7f05f6e44R436), right? Can we use some placeholder signal instead, e.g. `<Value>`, so that when an operator gets the ConfigMap it's obvious that this is only a placeholder?
The same goes for the other values as well.

Reply:
Hmm.. So are you suggesting the ConfigMap to have `advertise-client-urls` value like `<etcd-peer-service@default@2379>` ?

Reply:
No, like `<Service>@<Namespace>@<Port>`

Reply:
Please forget about my comments above. I just realized the logic behind it and that you actually need those values. Sorry for the confusion.

However, you need to pass `c.values.ClientServiceName` here.